tiledb/bioimg/converters/scale.py

```diff
@@ -0,0 +1,272 @@
+import multiprocessing
+from enum import Enum
+from functools import partial
+from multiprocessing import Pool
+from typing import List, Tuple
+
+from skimage.transform import resize
+
+import tiledb
+
+from .axes import Axes
+from .tiles import iter_tiles
+
+
+class ScaleMethod(Enum):
+    NEAREST = 0
+    LINEAR = 1
+
+
+class ScalerMode(Enum):
+    NON_PROGRESSIVE = 0
+    PROGRESSIVE = 1
+    CHUNKED_NON_PROGRESSIVE = 2
+    CHUNKED_PROGRESSIVE = 3
+
+
+class Scaler(object):
+    def __init__(
+        self, base: tiledb.array.DenseArray, axes: Axes, scale_factor: List[int]
+    ):
+        self._scale_factor = scale_factor
+        self._scale_factor_progressive = [float(scale_factor[0])]
+
+        for index in range(1, len(scale_factor)):
+            self._scale_factor_progressive.append(
+                scale_factor[index] / float(scale_factor[index - 1])
+            )
+
+        self._initial_shape = base.shape
+
+        self._resolutions = []
+
+        self._tiles = []
+        self._x_index = axes.dims.find("X")
+        self._y_index = axes.dims.find("Y")
+
+        for level in range(len(scale_factor)):
+            ratio = scale_factor[level]
+
+            shape = list(self._initial_shape)
+
+            shape[self._x_index] = int(
+                round(self._initial_shape[self._x_index] / float(ratio))
+            )
+            shape[self._y_index] = int(
+                round(self._initial_shape[self._y_index] / float(ratio))
+            )
```

Comment:
- No need for `float()`
- No need for `int()`: `round()` already returns an int (in Python 3)